src/k8s/request.js

```diff
@@ -238,14 +239,36 @@ const addCloudInit = (vm, basicSettings) => {
   }
 };
 
-const addDisk = (vm, diskSpec) => {
+const getDefaultDevice = (vm, deviceType, deviceName) =>
+  get(getDevices(vm), deviceType, []).find(device => device.name === deviceName);
+
+const getDomain = vm => {
   const domain = get(vm.spec.template.spec, 'domain', {});
-  const devices = get(domain, 'devices', {});
-  const disks = get(devices, 'disks', []);
-  disks.push(diskSpec);
-  devices.disks = disks;
-  domain.devices = devices;
   vm.spec.template.spec.domain = domain;
+  return domain;
+};
+
+const getDevices = vm => {
+  const devices = get(getDomain(vm), 'devices', {});
+  vm.spec.template.spec.domain.devices = devices;
```

Comment:
I think a condition would look better here, so we don't have to re-assign the variable when it's found. Applies for the other function as well.